api/v1alpha1/etcdcluster_types.go

```diff
@@ -54,6 +56,42 @@ type ProviderAutoConfig struct {
 }
 
 type ProviderCertManagerConfig struct {
+	// accepts a secret name with CABundle present
+	// this secret will be used to create an Issuer
+	// +optional
+	CABundleSecret string `json:"caBundleSecret,omitempty"`
```

Comment:
Please follow the best practice to add comment, this is a generic comment for all fields

https://go.dev/wiki/CodeReviewComments#comment-sentences

Also based on our discussion, the CABundleSecret isn't for Issuer, it's for etcd instance in case the secret generated by the provider (cert-manager) doesn't have CA.

Reply:
In case of cert-manager, it requires an Issuer to create a certificate secret. If it's a self-signed Issuer or there is an existing Issuer in the cluster(updated by user in `IssuerKind` and `IssuerName` ) then no `CABundleSecret` is required. 
But if there is no Issuer, this `CABundleSecret` is then used to create an Issuer which will in turn create the certificates. Ref: https://cert-manager.io/v1.6-docs/configuration/ca/#deployment

So, I think we can keep both the `CABundleSecret` and `IssuerKind, IssuerName` as mutually exclusive, like if one is already filled in, the other is not required. If both are filled, `CABundleSecret` will take the higher priority. WDTY?

Reply:
> But if there is no Issuer, this `CABundleSecret` is then used to create an Issuer

As we discussed previously, It isn't etcd-operator's responsibility to create an Issuer, nor should etcd-operator makes any changes to users' existing certificate system.

Reply:
can you deliver a separate PR to implement the solution 1 (reuse Config) mentioned above, also as we discussed in previous WG meeting and 1:1 meeting, so that we can compare the two solutions?

Reply:
Got it 👍